### DISCUSSION TITLE
Inject extra data

### DISCUSSION THREAD

**Discussion by johanboekhoven** [aminalaee/sqladmin#619] [Q&A]
Hi, have yet another question.

I have a standard schema/model and I'm creating a custom listview for it. Now in the custom view I create a custom table for data from another api. What do I need to override to along with the normal listview data also inject some custom data to be used in the template.
In a completely custom view I build the whole template response context where I can add whatever I like. But I'd rather use the default system so I can leverage all default behavior and just add a little bit of extra data to the template.

Looking through the ModeViewMeta source it's not apparent for me where the template calling context is being setup.

Thanks,
-Johan

**Comment by aminalaee** [comment]
Hey,
I think you should override `Admin.list` method here: https://github.com/aminalaee/sqladmin/blob/3b64b88c7460bfbf3f03faacb4e8970bda082f54/sqladmin/application.py#L415

And also the template probably needs some changing.

**Reply by johanboekhoven** [reply]
@aminalaee thanks for the quick response! 

I see, was looking at `async def list` in models.py but obviously it's in the base admin view.
So maybe I can add a metadata property to the context and have def list check for meta data in the modelView. Does that sounds reasonable?

**Reply by aminalaee** [reply]
I don't think you want anything from models.py because you mentioned you want to get data from another API right?
So I think you just fetch your data in this endpoint and show it? Or maybe you want to cache it in models.py and show it? That also works. Depends on your use-case.

**Reply by johanboekhoven** [reply]
@aminalaee I'm thinking too difficult once again, I can just add a property or function to the modelview and access it directly in the template. {{ model_view.my_property }} or function can just to the call and gather the data.

**Reply by aminalaee** [reply]
yeah if it's only one call that does make sense, just not sure how efficient it is to make those extra API calls, which I guess you have considered.

**Reply by johanboekhoven** [reply]
It's just one call, every so often on a slow external API anyway. No need for a highly optimized experience.
